runtime/near-vm-logic/src/tests/gas_counter.rs

```diff
@@ -15,7 +15,7 @@ fn test_dont_burn_gas_when_exceeding_attached_gas_limit() {
     let gas_limit = 10u64.pow(14);
 
     let mut logic_builder = VMLogicBuilder::default().max_gas_burnt(gas_limit * 2);
-    let mut logic = logic_builder.build_with_prepaid_gas(gas_limit);
+    let mut logic = logic_builder.build(get_context_with_prepaid_gas(gas_limit));
```

Comment:
FWIW we have been somewhat consistently moving towards adopting the builder approach more, rather than less. I wonder if that very same `VMLogicBuilder` could be adjusted to be able to construct a `TestVMLogic` too?

Reply:
Well, yes, VMLogicBuilder indeed builds a TestVMLogic.

The change here is that I’ve gotten rid of the `impl VMLogicBuilder`
block in this file as it felt out of place to me and introduced
`get_context_with_prepaid_gas` function.  I think the builder could be
improved since majority of `build` invocations are
`build(get_context(vec![], false))` but this seems like a change for
a different commit.